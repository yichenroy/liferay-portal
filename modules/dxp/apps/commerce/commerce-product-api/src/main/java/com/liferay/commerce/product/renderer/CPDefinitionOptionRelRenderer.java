/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 * <p>
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 * <p>
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.renderer;

import aQute.bnd.annotation.ProviderType;

import com.liferay.portal.kernel.exception.PortalException;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

/**
 * @author Marco Leo
 */
@ProviderType
public interface CPDefinitionOptionRelRenderer {

	public String render(
		long cpDefinitionId, HttpServletRequest httpServletRequest,
		HttpServletResponse httpServletResponse) throws PortalException;

}