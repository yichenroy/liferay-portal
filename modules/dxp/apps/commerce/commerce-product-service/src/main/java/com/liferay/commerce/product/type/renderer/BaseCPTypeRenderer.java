/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.type.renderer;

import com.liferay.commerce.product.type.CPTypeRenderer;
import com.liferay.portal.kernel.log.Log;
import com.liferay.portal.kernel.log.LogFactoryUtil;
import com.liferay.portal.kernel.model.PortletConstants;
import com.liferay.portal.kernel.portlet.PortletBag;
import com.liferay.portal.kernel.portlet.PortletBagPool;
import com.liferay.portal.kernel.util.Portal;
import com.liferay.portal.kernel.util.Validator;
import com.liferay.portal.kernel.util.WebKeys;

import java.io.IOException;

import javax.servlet.RequestDispatcher;
import javax.servlet.ServletContext;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.osgi.service.component.annotations.Reference;

/**
 * @author Marco Leo
 */
public abstract class BaseCPTypeRenderer implements CPTypeRenderer {

	public void renderJSP(
			HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse, String path)
		throws IOException {

		renderJSP(
			getServletContext(httpServletRequest), httpServletRequest,
			httpServletResponse, path);
	}

	public void renderJSP(
			ServletContext servletContext,
			HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse, String path)
		throws IOException {

		RequestDispatcher requestDispatcher =
			servletContext.getRequestDispatcher(path);

		try {
			requestDispatcher.include(httpServletRequest, httpServletResponse);
		}
		catch (ServletException se) {
			_log.error("Unable to render JSP " + path, se);

			throw new IOException("Unable to render " + path, se);
		}
	}

	public void setServletContext(ServletContext servletContext) {
		_servletContext = servletContext;
	}

	protected ServletContext getServletContext(
		HttpServletRequest httpServletRequest) {

		if (_servletContext != null) {
			return _servletContext;
		}

		String portletId = portal.getPortletId(httpServletRequest);

		if (Validator.isNotNull(portletId)) {
			String rootPortletId = PortletConstants.getRootPortletId(portletId);

			PortletBag portletBag = PortletBagPool.get(rootPortletId);

			return portletBag.getServletContext();
		}

		return (ServletContext)httpServletRequest.getAttribute(WebKeys.CTX);
	}

	@Reference
	protected Portal portal;

	private static final Log _log = LogFactoryUtil.getLog(
		BaseCPTypeRenderer.class);

	private ServletContext _servletContext;

}