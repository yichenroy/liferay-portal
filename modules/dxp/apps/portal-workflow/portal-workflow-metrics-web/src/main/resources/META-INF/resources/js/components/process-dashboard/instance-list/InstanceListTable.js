import InstanceListItem from './InstanceListItem';
import React from 'react';

export default class InstanceListTable extends React.Component {
	render() {
		const { items } = this.props;

		return (
			<div className="table-responsive">
				<table className="show-quick-actions-on-hover table table-fixed table-heading-nowrap table-hover table-list">
					<thead>
						<tr>
							<th
								className="text-center table-head-title"
								style={{ width: '10%' }}
							>
								{Liferay.Language.get('id')}
							</th>

							<th
								className="table-cell-expand table-head-title"
								style={{ width: '30%' }}
							>
								{Liferay.Language.get('item-subject')}
							</th>

							<th className="table-head-title" style={{ width: '25%' }}>
								{Liferay.Language.get('process-step')}
							</th>

							<th className="table-head-title" style={{ width: '20%' }}>
								{Liferay.Language.get('created-by')}
							</th>

							<th
								className="pr-4 table-head-title text-right"
								style={{ width: '15%' }}
							>
								{Liferay.Language.get('creation-date')}
							</th>
						</tr>
					</thead>

					<tbody>
						{items.map((item, index) => (
							<InstanceListItem {...item} key={index} />
						))}
					</tbody>
				</table>
			</div>
		);
	}
}