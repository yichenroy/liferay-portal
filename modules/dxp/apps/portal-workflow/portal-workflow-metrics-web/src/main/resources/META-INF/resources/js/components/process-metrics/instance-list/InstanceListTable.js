import InstanceListItem from './InstanceListItem';
import React from 'react';

const InstanceListTable = ({items}) => {
	return (
		<div className="table-responsive">
			<table
				className="show-quick-actions-on-hover table table-fixed table-heading-nowrap table-hover table-list"
				style={{minWidth: '1061px'}}
			>
				<thead>
					<tr>
						<th style={{width: '4%'}} />

						<th className="table-head-title" style={{width: '9%'}}>
							{Liferay.Language.get('id')}
						</th>

						<th
							className="table-cell-expand table-head-title"
							style={{width: '31%'}}
						>
							{Liferay.Language.get('item-subject')}
						</th>

						<th className="table-head-title" style={{width: '23%'}}>
							{Liferay.Language.get('process-step')}
						</th>

						<th className="table-head-title" style={{width: '15%'}}>
							{Liferay.Language.get('created-by')}
						</th>

						<th
							className="pr-4 table-head-title text-right"
							style={{width: '18%'}}
						>
							{Liferay.Language.get('creation-date')}
						</th>
					</tr>
				</thead>

				<tbody>
					{items.map((item, index) => (
						<InstanceListItem {...item} key={index} />
					))}
				</tbody>
			</table>
		</div>
	);
};

export default InstanceListTable;
