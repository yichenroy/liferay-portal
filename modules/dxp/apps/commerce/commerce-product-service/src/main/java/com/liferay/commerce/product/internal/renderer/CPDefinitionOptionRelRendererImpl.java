/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.internal.renderer;

import com.liferay.commerce.product.model.CPDefinitionOptionRel;
import com.liferay.commerce.product.model.CPDefinitionOptionValueRel;
import com.liferay.commerce.product.renderer.CPDefinitionOptionRelRenderer;
import com.liferay.commerce.product.service.CPDefinitionOptionRelService;
import com.liferay.dynamic.data.mapping.form.renderer.DDMFormRenderer;
import com.liferay.dynamic.data.mapping.form.renderer.DDMFormRenderingContext;
import com.liferay.dynamic.data.mapping.model.DDMForm;
import com.liferay.dynamic.data.mapping.model.DDMFormField;
import com.liferay.dynamic.data.mapping.model.DDMFormFieldOptions;
import com.liferay.dynamic.data.mapping.model.LocalizedValue;
import com.liferay.portal.kernel.exception.PortalException;
import com.liferay.portal.kernel.util.Portal;

import java.util.List;
import java.util.Locale;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.osgi.service.component.annotations.Component;
import org.osgi.service.component.annotations.Reference;

/**
 * @author Marco Leo
 */
@Component(immediate = true)
public class CPDefinitionOptionRelRendererImpl
	implements CPDefinitionOptionRelRenderer {

	@Override
	public String render(
			long cpDefinitionId, HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse)
		throws PortalException {

		DDMForm ddmForm = new DDMForm();

		Locale locale = _portal.getLocale(httpServletRequest);

		List<CPDefinitionOptionRel> cpDefinitionOptionRels =
			_cpDefinitionOptionRelService.
				getSkuContributorCPDefinitionOptionRels(cpDefinitionId);

		for (CPDefinitionOptionRel cpDefinitionOptionRel :
				cpDefinitionOptionRels) {

			List<CPDefinitionOptionValueRel> cpDefinitionOptionValueRels =
				cpDefinitionOptionRel.getCPDefinitionOptionValueRels();

			DDMFormField ddmFormField = new DDMFormField(
				String.valueOf(cpDefinitionOptionRel.
					getCPDefinitionOptionRelId()),
				cpDefinitionOptionRel.getDDMFormFieldTypeName());

			if (!cpDefinitionOptionValueRels.isEmpty()) {
				DDMFormFieldOptions ddmFormFieldOptions =
					new DDMFormFieldOptions();

				for (CPDefinitionOptionValueRel cpDefinitionOptionValueRel :
						cpDefinitionOptionValueRels) {

					ddmFormFieldOptions.addOptionLabel(
						String.valueOf(cpDefinitionOptionValueRel.
							getCPDefinitionOptionValueRelId()),
						locale, cpDefinitionOptionValueRel.getTitle(locale));
				}

				ddmFormField.setDDMFormFieldOptions(ddmFormFieldOptions);
			}

			LocalizedValue localizedValue = new LocalizedValue(locale);

			localizedValue.addString(
				locale, cpDefinitionOptionRel.getName(locale));

			ddmFormField.setLabel(localizedValue);

			ddmFormField.setRequired(true);

			ddmForm.addDDMFormField(ddmFormField);
		}

		DDMFormRenderingContext ddmFormRenderingContext =
			new DDMFormRenderingContext();

		ddmFormRenderingContext.setContainerId("options");
		ddmFormRenderingContext.setHttpServletRequest(httpServletRequest);
		ddmFormRenderingContext.setHttpServletResponse(httpServletResponse);
		ddmFormRenderingContext.setLocale(locale);

		return _ddmFormRenderer.render(ddmForm, ddmFormRenderingContext);
	}

	@Reference
	private CPDefinitionOptionRelService _cpDefinitionOptionRelService;

	@Reference
	private DDMFormRenderer _ddmFormRenderer;

	@Reference
	private Portal _portal;

}