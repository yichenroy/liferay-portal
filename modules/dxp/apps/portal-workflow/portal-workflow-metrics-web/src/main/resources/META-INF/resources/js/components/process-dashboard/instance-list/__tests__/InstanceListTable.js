import InstanceListTable from '../InstanceListTable';
import React from 'react';
import renderer from 'react-test-renderer';
import { MockRouter as Router } from '../../../../test/mock/MockRouter';

test('Should render component', () => {
	const data = [
		{
			assetName: 'Item Subject Test',
			assetType: 'Process',
			createdBy: 'User Test',
			creationDate: new Date('2019', '01', '01'),
			id: 12351,
			processSteps: ['Step 1', 'Step 2', 'Step 3'],
			status: 'on-time'
		},
		{
			assetName: 'Item Subject Test 2',
			assetType: 'Process',
			createdBy: 'User Test',
			creationDate: new Date('2019', '01', '02'),
			id: 12351,
			processSteps: ['Step 1', 'Step 2'],
			status: 'overdue'
		},
		{
			assetName: 'Item Subject Test 3',
			assetType: 'Process',
			createdBy: 'User Test',
			creationDate: new Date('2019', '01', '03'),
			id: 12351,
			status: 'no-sla-counting'
		}
	];

	const component = renderer.create(
		<Router client={fetch(data)}>
			<InstanceListTable items={data} />
		</Router>
	);

	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});