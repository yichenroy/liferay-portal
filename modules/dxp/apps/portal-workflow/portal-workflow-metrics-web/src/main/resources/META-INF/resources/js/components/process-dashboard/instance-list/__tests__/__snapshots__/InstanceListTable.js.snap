// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render component 1`] = `
<div
  className="table-responsive"
>
  <table
    className="show-quick-actions-on-hover table table-fixed table-heading-nowrap table-hover table-list"
  >
    <thead>
      <tr>
        <th
          style={
            Object {
              "width": "5%",
            }
          }
        />
        <th
          className="table-head-title"
          style={
            Object {
              "width": "10%",
            }
          }
        >
          id
        </th>
        <th
          className="table-cell-expand table-head-title"
          style={
            Object {
              "width": "30%",
            }
          }
        >
          item-subject
        </th>
        <th
          className="table-head-title"
          style={
            Object {
              "width": "25%",
            }
          }
        >
          process-step
        </th>
        <th
          className="table-head-title"
          style={
            Object {
              "width": "15%",
            }
          }
        >
          created-by
        </th>
        <th
          className="pr-4 table-head-title text-right"
          style={
            Object {
              "width": "15%",
            }
          }
        >
          creation-date
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <span
            className="mr-3 sticker sticker-sm bg-success-light"
          >
            <span
              className="inline-item"
            >
              <svg
                className="lexicon-icon lexicon-icon-check-circle text-success"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<use xlink:href=\\"http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#check-circle\\" />",
                  }
                }
                focusable="false"
                role="presentation"
              />
            </span>
          </span>
        </td>
        <td
          className="lfr-title-column table-title"
        >
          <a
            data-target="#instanceDetailModal"
            data-toggle="modal"
            href="javascript:;"
            onClick={[Function]}
            tabIndex="-1"
          >
            <strong>
              12351
            </strong>
          </a>
        </td>
        <td>
          Process: Item Subject Test
        </td>
        <td>
          Step 1, Step 2, Step 3
        </td>
        <td>
          User Test
        </td>
        <td
          className="pr-4 text-right"
        >
          Feb 01, 12:00 AM
        </td>
      </tr>
      <tr>
        <td>
          <span
            className="mr-3 sticker sticker-sm bg-danger-light"
          >
            <span
              className="inline-item"
            >
              <svg
                className="lexicon-icon lexicon-icon-exclamation-circle text-danger"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<use xlink:href=\\"http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#exclamation-circle\\" />",
                  }
                }
                focusable="false"
                role="presentation"
              />
            </span>
          </span>
        </td>
        <td
          className="lfr-title-column table-title"
        >
          <a
            data-target="#instanceDetailModal"
            data-toggle="modal"
            href="javascript:;"
            onClick={[Function]}
            tabIndex="-1"
          >
            <strong>
              12351
            </strong>
          </a>
        </td>
        <td>
          Process: Item Subject Test 2
        </td>
        <td>
          Step 1, Step 2
        </td>
        <td>
          User Test
        </td>
        <td
          className="pr-4 text-right"
        >
          Feb 02, 12:00 AM
        </td>
      </tr>
      <tr>
        <td>
          <span
            className="mr-3 sticker sticker-sm bg-info-light"
          >
            <span
              className="inline-item"
            >
              <svg
                className="lexicon-icon lexicon-icon-hr text-info"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<use xlink:href=\\"http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#hr\\" />",
                  }
                }
                focusable="false"
                role="presentation"
              />
            </span>
          </span>
        </td>
        <td
          className="lfr-title-column table-title"
        >
          <a
            data-target="#instanceDetailModal"
            data-toggle="modal"
            href="javascript:;"
            onClick={[Function]}
            tabIndex="-1"
          >
            <strong>
              12351
            </strong>
          </a>
        </td>
        <td>
          Process: Item Subject Test 3
        </td>
        <td>
          completed
        </td>
        <td>
          User Test
        </td>
        <td
          className="pr-4 text-right"
        >
          Feb 03, 12:00 AM
        </td>
      </tr>
      <tr>
        <td />
        <td
          className="lfr-title-column table-title"
        >
          <a
            data-target="#instanceDetailModal"
            data-toggle="modal"
            href="javascript:;"
            onClick={[Function]}
            tabIndex="-1"
          >
            <strong>
              12351
            </strong>
          </a>
        </td>
        <td>
          Process: Item Subject Test 4
        </td>
        <td>
          completed
        </td>
        <td>
          User Test
        </td>
        <td
          className="pr-4 text-right"
        >
          Feb 04, 12:00 AM
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
